src/routes/run.rs

```diff
@@ -100,21 +121,35 @@ async fn find_by_id(req: HttpRequest, pool: web::Data<db::DbPool>) -> impl Respo
         }
     })
     .await
-    // If there is no error, return a response with the retrieved data
-    .map(|results| HttpResponse::Ok().json(results))
-    .map_err(|e| {
-        error!("{}", e);
-        match e {
-            // If no run is found, return a 404
-            BlockingError::Error(diesel::NotFound) => HttpResponse::NotFound().json(ErrorBody {
-                title: "No run found".to_string(),
-                status: 404,
-                detail: "No run found with the specified ID".to_string(),
-            }),
-            // For other errors, return a 500
-            _ => default_500(&e),
+    {
+        Ok(results) => {
+            // Return it as csvs if the user requested that
+            if return_csvs {
+                // Build csv files from results
+                let zip_bytes: Vec<u8> = get_bytes_for_csv_zip_from_runs(&vec![results])?;
```

Comment:
You're correct that the purpose of zipping the CSVs is because it's easier (on the carrot end and the cli end) to handle one zip file as opposed to several CSVs.  I'm not sure there's even a convention for returning multiple files in one http response.  Http conventions might not be into that. 

As for the JSON end of things, with few exceptions, all the API endpoints in CARROT accept and return JSON because that's a best practice for REST APIs.  Zipping the JSON would be kinda weird because it's not really intended to be treated as a file (although it can be).